classes/install/Upgrade.inc.php

```diff
@@ -127,29 +126,41 @@ public function setFileUploader()
     public function setFileName()
     {
         $journalDao = DAORegistry::getDAO('JournalDAO'); /* @var $journalDao JournalDAO */
-        $submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
 
         $contexts = $journalDao->getAll();
         while ($context = $contexts->next()) {
             $collector = Repo::submission()->getCollector()->filterByContextIds([$context->getId()]);
             $submissionIds = Repo::submission()->getIds($collector);
             foreach ($submissionIds as $submissionId) {
-                $submissionFilesIterator = Services::get('submissionFile')->getMany([
-                    'submissionIds' => [$submissionId],
-                    'includeDependentFiles' => true,
-                ]);
+                $submissionFileCollector = Repo::submissionFiles()
+                    ->getCollector()
+                    ->filterBySubmissionIds([$submissionId])
+                    ->filterByIncludeDependentFiles(true);
+
+                $submissionFilesIterator = Repo::submissionFiles()
+                    ->getMany($submissionFileCollector);
+
                 foreach ($submissionFilesIterator as $submissionFile) {
-                    $reviewStage = $submissionFile->getFileStage() == SubmissionFile::SUBMISSION_FILE_REVIEW_FILE ||
-                        $submissionFile->getFileStage() == SubmissionFile::SUBMISSION_FILE_REVIEW_ATTACHMENT ||
-                        $submissionFile->getFileStage() == SubmissionFile::SUBMISSION_FILE_REVIEW_REVISION;
-                    if (!$submissionFile->getName(AppLocale::getPrimaryLocale())) {
-                        if ($reviewStage) {
-                            $submissionFile->setName($submissionFile->_generateName(true), AppLocale::getPrimaryLocale());
-                        } else {
-                            $submissionFile->setName($submissionFile->_generateName(), AppLocale::getPrimaryLocale());
-                        }
+                    if ($submissionFile->getName(AppLocale::getPrimaryLocale())) {
+                        continue;
                     }
-                    $submissionFileDao->updateObject($submissionFile);
+
+                    $itsOnReviewStage = (bool) (
+                        $submissionFile->itsOnReviewFileStage() ||
+                        $submissionFile->itsOnReviewAttachmentStage() ||
+                        $submissionFile->itsOnReviewRevisionStage()
+                    );
+
+                    $data = [
+                        'name' => [
+                            AppLocale::getPrimaryLocale() => $submissionFile->_generateName($itsOnReviewStage),
+                        ]
+                    ];
+
+                    Repo::submissionFiles()->edit(
+                        $submissionFile,
+                        $data
+                    );
```

Comment:
And FWIW, it's best to avoid using DAOs/repositories/collectors in upgrade code -- better to use DB::execute type calls instead. This upgrade script is written to work with 3.0.0-x code, and the queries in the DAO/repository/collector code will be written with the current (3.4.0) database schema in mind. If the two are different -- or become different later -- the upgrade script will fail.